alarmdecoder/decoder.py

```diff
@@ -261,19 +264,21 @@ def open(self, baudrate=None, no_reader_thread=False):
         :type no_reader_thread: bool
         """
         self._wire_events()
-        self._device.open(baudrate=baudrate, no_reader_thread=no_reader_thread)
+        try:
+            self._device.open(baudrate=baudrate,
+                              no_reader_thread=no_reader_thread)
+        except:
+            self._unwire_events
+            raise
 
         return self
 
     def close(self):
         """
         Closes the device.
         """
-        if self._device:
```

Comment:
Agree. The destruction of the object that was passed in from the parent was an error as well as the tests for its validity.